### DISCUSSION TITLE
Is it possible to run Vertipaq Analyzer overall datasets in a workspace?

### DISCUSSION THREAD

**Discussion by pawarbi** [DaxStudio/DaxStudio#1050] [Q&A]
Is it possible to run Vertipaq Analyzer / View Metrics on all datasets in a Premium workspace ? I can select them manually but wondering if it's possible to do it programmatically ? I tried running `$SYSTEM.DISCOVER_MEMORYUSAGE` DMV on the published dataset using XMLA in DAX Studio but got `User does not have permission to call the Discover method` though I am the admin of the workspace and the author of the dataset. Any thoughts? Thank you.

**Comment by dgosbell** [comment]
Yes, you could run it programmatically. Vertipaq Analyzer is available as a separate library which extracts the information and reads/writes the vpax files (see https://github.com/sql-bi/VertiPaq-Analyzer ). It is also available on nuget as a series of packages. So you could build a powershell script or c# app to loop over all the datasets in a workspace and generate a vpax file for each one.

**Reply by pawarbi** [reply]
Thanks Darren. Can you please point me to any existing examples, if any, on how to load , connect Vertipaq analyzer using PowerShell?I know how to use Vertipaq analyzer using DAX studio, Excel and TE. Not sure about PowerShell. Thanks
